### PR TITLE
Add metamodels to Gradle sources

### PR DESCRIPTION
**Pull request by nickclyde**
## Related Issue or Background Info

- VS Code currently throws a bunch of errors like this for metamodel imports used in our JPA Specifications:
image

## Changes Proposed

- Add `build/generated/sources/annotationProcessor/java/main` to our `sourceSets` in `build.gradle`. This gets rid of the errors
- Also combined the split `test` sections in `build.gradle` to get rid of a warning about them being split

## Additional Information

- See this [StackOverflow post](https://stackoverflow.com/questions/28345705/how-can-i-add-a-generated-source-folder-to-my-source-path-in-gradle)

## Screenshots / Demos
image

### UI
- [ ] Any changes to the UI/UX are approved by design 
- [ ] Any new or updated content (e.g. error messages) are approved by design 

### Testing
- [ ] Includes a summary of what a code reviewer should verify

### Changes are Backwards Compatible
- [ ] Database changes are submitted as a separate PR
  - [ ] Any new tables that do not contain PII are accompanied by a GRANT SELECT to the no-PHI user
  - [ ] Any changes to tables that have custom no-PHI views are accompanied by changes to those views
        (including re-granting permission to the no-PHI user if need be)
  - [ ] Liquibase rollback has been tested locally using `./gradlew liquibaseRollbackSQL` or `liquibaseRollback`
  - [ ] Each new changeset has a corresponding [tag](https://docs.liquibase.com/change-types/community/tag-database.html)
- [ ] GraphQL schema changes are backward compatible with older version of the front-end

### Security
- [ ] Changes with security implications have been approved by a security engineer (changes to  authentication, encryption, handling of PII, etc.)
- [ ] Any dependencies introduced have been vetted and discussed

## Cloud
- [ ] DevOps team has been notified if PR requires ops support
- [ ] If there are changes that cannot be tested locally, this has been deployed to our Azure `test`, `dev`, or `pentest` environment for verification

### CODE REVIEW

File: backend/build.gradle [hunk 1]
```diff
@@ -214,9 +201,26 @@ task testDbReset(type: Exec) {
 }
 
 test {
+    // people sometimes export this to run local dev more easily: don't let it break the tests
```
Comment (bmaden-usds) [inline backend/build.gradle:204]: Thanks for cleaning this up!

File: backend/config/checkstyle/suppressions.xml [hunk 2]
```diff
@@ -38,5 +38,37 @@
 	<suppress checks="FileTabCharacter" files="src/main/java/gov/cdc/usds/simplereport/db/repository/TestEventRepository.java" />
 	<suppress checks="FileTabCharacter" files="src/main/java/gov/cdc/usds/simplereport/service/TestOrderService.java" />
 	<suppress checks="FileTabCharacter" files="src/main/java/gov/cdc/usds/simplereport/service/UploadService.java" />
+	<suppress checks="FileTabCharacter" files="build/generated/sources/annotationProcessor/java/main/gov/cdc/usds/simplereport/db/model/ApiAuditEvent_.java" />
```
Comment (bmaden-usds) [inline backend/config/checkstyle/suppressions.xml:41]: No way to glob these?

**Review by bmaden-usds — APPROVED** [review]
LGTM, but I'm not a VSCode user!

File: backend/config/checkstyle/suppressions.xml (hunk 2 above)
Comment (nickclyde) [reply backend/config/checkstyle/suppressions.xml:41]: Not that I could figure out :cry:

**Pull request merged**